pkg/results/perf_results.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2020 Coinbase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package results
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/coinbase/rosetta-cli/configuration"
+	"github.com/coinbase/rosetta-sdk-go/utils"
+	"github.com/olekukonko/tablewriter"
+)
+
+// Output writes *CheckPerfResults to the provided
+// path.
+func (c *CheckPerfStats) Output(path string) {
+	if len(path) > 0 {
+		writeErr := utils.SerializeAndWrite(path, c)
+		if writeErr != nil {
+			log.Printf("%s: unable to save results\n", writeErr.Error())
+		}
+	}
+}
+
+type CheckPerfRawStats struct {
+	BlockEndpointTotalTime          time.Duration
+	BlockEndpointNumErrors          int64
+	AccountBalanceEndpointTotalTime time.Duration
+	AccountBalanceNumErrors         int64
+}
+
+// CheckPerfStats contains interesting stats that
+// are counted while running the check:perf.
+type CheckPerfStats struct {
+	StartBlock                          int64 `json:"start_block"`
+	EndBlock                            int64 `json:"end_block"`
+	NumTimesHitEachEndpoint             int   `json:"num_times_hit_each_endpoint"`
+	AccountBalanceEndpointAverageTimeMs int64 `json:"account_balance_endpoint_average_time_ms"`
+	AccountBalanceEndpointTotalTimeMs   int64 `json:"account_balance_endpoint_total_time_ms"`
+	AccountBalanceEndpointNumErrors     int64 `json:"account_balance_endpoint_num_errors"`
+	BlockEndpointAverageTimeMs          int64 `json:"block_endpoint_average_time_ms"`
+	BlockEndpointTotalTimeMs            int64 `json:"block_endpoint_total_time_ms"`
+	BlockEndpointNumErrors              int64 `json:"block_endpoint_num_errors"`
+}
+
+// Print logs CheckPerfStats to the console.
+func (c *CheckPerfStats) Print() {
+	table := tablewriter.NewWriter(os.Stdout)
+	table.SetRowLine(true)
+	table.SetRowSeparator("-")
+	table.SetHeader([]string{"check:perf Stats", "Description", "Value"})
+	table.Append([]string{"Start Block", "Start Block", strconv.FormatInt(c.StartBlock, 10)})
+	table.Append([]string{"End Block", "EndBlock", strconv.FormatInt(c.EndBlock, 10)})
```

Comment:
It may be better for the descriptions to be "The starting block." and "The ending block." If anything, at least "EndBlock" should be "End Block", to match the formatting in line 67.

```suggestion
	table.Append([]string{"Start Block", "Start Block", strconv.FormatInt(c.StartBlock, 10)})
	table.Append([]string{"End Block", "End Block", strconv.FormatInt(c.EndBlock, 10)})
```